### PR TITLE
Add create timetable tests

### PR DESCRIPTION
**Pull request by majimearun**

### CODE REVIEW

File: src/controllers/timetable/createTimetable.test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    test("Test if error issues has a message", () => {
+      expect(response?.body).toHaveProperty("message");
+    });
+
+    test("Test if response error is correct", () => {
+      expect(response?.body.message).toEqual("unregistered user");
+    });
+  });
+
+  describe("Test createTimetable as single degree student", () => {
```
Comment (skoriop) [inline src/controllers/timetable/createTimetable.test.ts:122]: Maybe change it to "Test createTimetable 200 (single degree student)" to match the style of the others

File: src/controllers/timetable/createTimetable.test.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    test("Test if returns truthy response", () => {
+      expect(response?.body).toBeTruthy();
+    });
+
+    test("Test if returns status 201", () => {
+      expect(response?.status).toEqual(201);
+    });
+  });
+
+  describe("Test createTimetable as dual degree student", () => {
```
Comment (skoriop) [inline src/controllers/timetable/createTimetable.test.ts:139]: Same point about the name

**Review by skoriop — CHANGES_REQUESTED** [review]
I think you should also add two more tests in the last two test groups:
- Check if response message matches what is expected
- Check if the timetable was actually created (using a DB query)

Other than that (and the naming thing), LGTM.

File: src/controllers/timetable/createTimetable.test.ts (hunk 1 above)
Comment (majimearun) [reply src/controllers/timetable/createTimetable.test.ts:122]: done

File: src/controllers/timetable/createTimetable.test.ts (hunk 2 above)
Comment (majimearun) [reply src/controllers/timetable/createTimetable.test.ts:139]: done

File: src/controllers/timetable/createTimetable.test.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+    test("Test if returns truthy response", () => {
+      expect(response?.body).toBeTruthy();
+    });
+
+    test("Test if returns status 201", () => {
+      expect(response?.status).toEqual(201);
+    });
+
+    test("Test if response error is correct", () => {
```
Comment (skoriop) [inline src/controllers/timetable/createTimetable.test.ts:149]: it's just a message, not an error

File: src/controllers/timetable/createTimetable.test.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
```
Comment (skoriop) [inline src/controllers/timetable/createTimetable.test.ts:179]: Same as previous

File: src/controllers/timetable/createTimetable.test.ts [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    test("Test if error issues has a message", () => {
+      expect(response?.body).toHaveProperty("message");
+    });
+
+    test("Test if response error is correct", () => {
+      expect(response?.body.message).toEqual("unregistered user");
+    });
+  });
+
+  describe("Test createTimetable 200 (as single degree student)", () => {
```
Comment (skoriop) [inline src/controllers/timetable/createTimetable.test.ts:132]: It should be 201 not 200 (my bad, sorry)

File: src/controllers/timetable/createTimetable.test.ts [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+    test("Test if timetable is created", async () => {
+      timetable = await timetableRepository
+        .createQueryBuilder("timetable")
+        .where("timetable.authorId = :id", { id: uuids[0] })
+        .getOne();
+      expect(timetable).toBeTruthy();
+    });
+  });
+
+  describe("Test createTimetable 200 (as dual degree student)", () => {
```
Comment (skoriop) [inline src/controllers/timetable/createTimetable.test.ts:162]: Same as previous

File: src/controllers/timetable/createTimetable.test.ts [hunk 7]
```diff
@@ -21,15 +23,16 @@ describe("Test createTimetable", () => {
       batch: 2021,
       name: "UVW XYZ",
       degrees: ["A7"] as degreeEnum[],
-      email: "f20210000@hyderabad.bits-pilani.ac.in",
+      email: "f202100000@hyderabad.bits-pilani.ac.in",
```
Comment (skoriop) [inline src/controllers/timetable/createTimetable.test.ts:26]: Any specific reason why?

**Review by skoriop — CHANGES_REQUESTED** [review]

File: src/controllers/timetable/createTimetable.test.ts (hunk 3 above)
Comment (majimearun) [reply src/controllers/timetable/createTimetable.test.ts:149]: i hate copy paste

File: src/controllers/timetable/createTimetable.test.ts (hunk 5 above)
Comment (majimearun) [reply src/controllers/timetable/createTimetable.test.ts]: me being an idiota did not realise as well lmao

File: src/controllers/timetable/createTimetable.test.ts (hunk 7 above)
Comment (majimearun) [reply src/controllers/timetable/createTimetable.test.ts:26]: your delete had the same user emails i think, so for the unique key thing

Comment (skoriop) [reply src/controllers/timetable/createTimetable.test.ts:26]: hmm alright

**Review by skoriop — APPROVED** [review]
LGTM

**Pull request merged**